### PR TITLE
New design (UI) - Choose account dialog

### PR DESCRIPTION
**Pull request by Shagequi**
Hi, 

This PR is the new Dialog to choose and manage your accounts (instead of using Drawer).

| Device | Lightmode | Darkmode |
|---|---|---|
|Phone - few accounts|Capture d’écran 2020-05-20 à 13 02 00|Capture d’écran 2020-05-20 à 13 02 34
Phone - many accounts|Capture d’écran 2020-05-20 à 13 03 12|Capture d’écran 2020-05-20 à 13 02 48|
Tablet - few accounts|Capture d’écran 2020-05-20 à 13 03 26|Capture d’écran 2020-05-20 à 13 03 50

The general behaviour is the following : 

demo_android_gif_ft_2

You're able to : 
- Manage accounts
- Add an account
- "Take control" of an account

The number of accounts doesn't affect the functionning of the dialog. 

Thank you.

Based on : https://github.com/nextcloud/android/issues/5207

### Testing 
Writing tests is very important. Please try to write some tests for your PR. 
If you need help, please do not hesitate to ask in this PR for help.

[unit tests](https://github.com/nextcloud/android/blob/master/CONTRIBUTING.md#unit-tests)
[instrumented tests](https://github.com/nextcloud/android/blob/master/CONTRIBUTING.md#instrumented-tests)
[UI tests](https://github.com/nextcloud/android/blob/master/CONTRIBUTING.md#ui-tests)

- [X] Tests written, or not not needed

### CODE REVIEW

File: src/main/java/com/owncloud/android/ui/adapter/UserListAdapter.java [hunk 1]
```diff
@@ -68,13 +67,14 @@
...
     private boolean showAddAccount;
+    private boolean showDotsMenu;
 
     public UserListAdapter(BaseActivity context,
                            UserAccountManager accountManager,
                            List<UserListItem> values,
                            Drawable tintedCheck,
                            ClickListener clickListener,
-                           boolean showAddAccount) {
+                           boolean showAddAccount, boolean showDotsMenu) {
```
Comment (AndyScherzinger) [inline src/main/java/com/owncloud/android/ui/adapter/UserListAdapter.java:77]: should be on a new line

File: src/main/java/com/owncloud/android/ui/dialog/MultipleAccountsDialog.java [hunk 2]
```diff
@@ -95,7 +95,7 @@ public Dialog onCreateDialog(Bundle savedInstanceState) {
                                                       getAccountListItems(),
                                                       tintedCheck,
                                                       this,
-                                                      false);
+                                                      false, false);
```
Comment (AndyScherzinger) [inline src/main/java/com/owncloud/android/ui/dialog/MultipleAccountsDialog.java:98]: should be on a new line

File: src/main/res/drawable/ic_check_circle.xml [hunk 3]
```diff
@@ -1,9 +1,5 @@
-<vector xmlns:android="http://schemas.android.com/apk/res/android"
-    android:width="24dp"
-    android:height="24dp"
-    android:viewportWidth="24"
-    android:viewportHeight="24">
-    <path
-        android:fillColor="#FF000000"
-        android:pathData="M12,2C6.48,2 2,6.48 2,12s4.48,10 10,10 10,-4.48 10,-10S17.52,2 12,2zM10,17l-5,-5 1.41,-1.41L10,14.17l7.59,-7.59L19,8l-9,9z" />
+<vector android:height="24dp" android:tint="#4995FB"
```
Comment (AndyScherzinger) [inline src/main/res/drawable/ic_check_circle.xml:1]: could you hit "formatting" for the file? :)

File: src/main/res/layout/account_item.xml [hunk 4]
```diff
@@ -1,10 +1,12 @@
 <?xml version="1.0" encoding="utf-8"?>
+
```
Comment (AndyScherzinger) [inline src/main/res/layout/account_item.xml:2]: unneeded linebreak

File: src/main/res/layout/dialog_choose_account.xml [hunk 5]
```diff
@@ -0,0 +1,97 @@
+<!--
+  Nextcloud Android client application
+
+  Copyright (C) 2020 Infomaniak Network SA
```
Comment (AndyScherzinger) [inline src/main/res/layout/dialog_choose_account.xml:4]: feel free to also add yourself as an author ❤️

File: src/main/res/values/colors.xml [hunk 6]
```diff
@@ -87,4 +87,7 @@
     <color name="appbar">@android:color/white</color>
     <color name="fontAppbar">#666666</color>
     <color name="fontSecondaryAppbar">#A5A5A5</color>
+
+    <!-- Account selection -->
+    <color name="account_separator">#E0E0E0</color>
```
Comment (AndyScherzinger) [inline src/main/res/values/colors.xml:92]: should probably match `list_divider_background` while that color seems to be slightly to dark (at least on dark theme). So I'd say we should unify this and only have a single divider color definition used for all dividers.

I do like your choice `#E0E0E0`.

File: src/main/res/values/dims.xml [hunk 7]
```diff
@@ -145,4 +145,7 @@
     <dimen name="button_corner_radius">24dp</dimen>
     <integer name="media_grid_width">4</integer>
     <dimen name="copy_internal_link_padding">10dp</dimen>
+    <dimen name="account_action_button_margin">8dp</dimen>
```
Comment (AndyScherzinger) [inline src/main/res/values/dims.xml:148]: this is basically `standard_half_margin` and could be used instead

**Comment by AndyScherzinger** [comment]
@Shagequi just some very minor and very picky comments. Code looks good, testing succesful.
One thing I am unsure about but maybe @tobiasKaminsky or @jancborchardt can answer this. The checkmark icon used for marking the active account is always "blue". Is this un porpuse or should the icon be tinted based on the server-side primary color? I guess the later but don't know whatr you folks might have discussed already.

File: src/main/res/layout/dialog_choose_account.xml (hunk 5 above)
Comment (Shagequi) [reply src/main/res/layout/dialog_choose_account.xml:4]: Not needed ;)

File: src/main/res/values/colors.xml (hunk 6 above)
Comment (Shagequi) [reply src/main/res/values/colors.xml:92]: For now, I don't touch

File: src/main/java/com/owncloud/android/ui/adapter/UserListAdapter.java (hunk 1 above)
Comment (Shagequi) [reply src/main/java/com/owncloud/android/ui/adapter/UserListAdapter.java]: Fixed :)

File: src/main/res/layout/account_item.xml (hunk 4 above)
Comment (Shagequi) [reply src/main/res/layout/account_item.xml]: Fixed :)

File: src/main/res/values/dims.xml (hunk 7 above)
Comment (Shagequi) [reply src/main/res/values/dims.xml]: Fixed :)

File: src/main/res/drawable/ic_check_circle.xml (hunk 3 above)
Comment (Shagequi) [reply src/main/res/drawable/ic_check_circle.xml]: Fixed :)

File: src/main/java/com/owncloud/android/ui/dialog/MultipleAccountsDialog.java (hunk 2 above)
Comment (Shagequi) [reply src/main/java/com/owncloud/android/ui/dialog/MultipleAccountsDialog.java]: Fixed :)

File: src/main/res/values/colors.xml (hunk 6 above)
Comment (AndyScherzinger) [reply src/main/res/values/colors.xml:92]: fine with me 👍

**Comment by JorisBodin** [comment]
Now it's better ? :)
|Light|Dark|
|---|---|
|light|dark|

I use background color. But i prefere if we use appbar color.

**Comment by tobiasKaminsky** [comment]
I will look into detail on Friday, but two questions for now
- what do we do with this:

This can be removed, or? @jancborchardt 

- 
 

Is this a different font? "Manage" looks a bit different on the left (new) vs right (old)?

**Comment by AndyScherzinger** [comment]
> I use background color. But i prefere if we use appbar color.

@JorisBodin can't say if it would be the appbar color but yeah I also vote to not use the background color since this doesn't work for the dark theme due to the lack of elevation, if that matches the appbar-color then yeah, sounds perfect to me. 👍 @jancborchardt what do you think? lighter background color for the pop-up on the dark theme? (actually how it should be done defined by material design 😃 )

**Comment by AndyScherzinger** [comment]
> what do we do with this:

IMHO needs to be removed but via another PR

> Is this a different font? "Manage" looks a bit different on the left (new) vs right (old)?

Seems like, that looks like the newer material font while the "old one" is Roboto

**Comment by jancborchardt** [comment]
> @jancborchardt what do you think? lighter background color for the pop-up on the dark theme? (actually how it should be done defined by material design smiley )

Sounds good, yes! :) 



> > what do we do with this:
> 
> IMHO needs to be removed but via another PR

Agreed here as well. :)

**Comment by AndyScherzinger** [comment]
@Shagequi any ideas about the font issue? https://github.com/nextcloud/android/pull/6116#issuecomment-631604621

**Comment by Shagequi** [comment]
@AndyScherzinger @tobiasKaminsky

Hi ! This "issue" is present on many dialogs on the project : 

Capture d’écran 2020-05-25 à 19 00 47
Capture d’écran 2020-05-25 à 19 01 09

We can indeed see the letter spacing is on the same on Buttons in Dialog, and on "standard" labels. I don't know why these font look like this :/, but to be perfectly honest, I wouldn't dwell on it. 
Some tasks (such as the redesign of the Drawer) await the merge of this important PR and we need to move forward 😊 

If you don't like the font, there's always a way to make an issue a little later, and correct it once the whole design is redone. But but from my point of view, this PR is functional ^_^

**Comment by jancborchardt** [comment]
Yeah, sounds like the font issue is separate from this pull request. :)

**Comment by JorisBodin** [comment]
The font is different, because we now use new Material components.
It's the new material button, so they're using automaticaly the new material font ...

> 

These buttons will be remove, so we won't see the difference :P

**Comment by AndyScherzinger** [comment]
Thanks for the explanation and agreed, another, separate issue 👍

**Comment by tobiasKaminsky** [comment]
Lint increased:
```
../../../src/main/res/layout/dialog_choose_account.xml:23: Possible overdraw: Root element paints background @color/bg_default with a theme that also paints a background (inferred theme is @style/Theme.ownCloud.Toolbar)
```

```
../../../src/main/res/values/colors.xml:92: The resource R.color.account_separator appears to be unused
```

**Comment by JorisBodin** [comment]
@tobiasKaminsky it's fixed :)

**Pull request merged**